addon/services/address-register.js

```diff
@@ -0,0 +1,128 @@
+import Service from '@ember/service';
+import fetch from 'fetch';
+
+export default class AddressRegisterService extends Service {
+  /**
+   * Suggests addresses from a given string.
+   * Via the API of loc.geopunt.be
+   *
+   * @param {string} fuzzyString The searched address, a fuzzy search will be performed
+   * @returns {AddressSuggestion}
+   */
+  async suggest(fuzzyString) {
+    const results = await (
+      await fetch(`/adresses-register/search?query=${fuzzyString}`)
+    ).json();
+    return results.adressen.map(function (result) {
+      return new AddressSuggestion({
+        id: result.ID,
+        street: result.Thoroughfarename,
+        housenumber: result.Housenumber,
+        zipCode: result.Zipcode,
+        municipality: result.Municipality,
+        fullAddress: result.FormattedAddress,
+      });
+    });
+  }
+
+  /**
+   * Finds the addresses matching the suggestion and get all its properties (including its URI)
+   * Via the API of https://basisregisters.vlaanderen.be/api
+   *
+   * @param {AddressSuggestion} suggestion The suggested address
+   * @returns {Array[Address]}
+   */
+  async findAll(suggestion) {
+    let addresses = [];
+    if (!suggestion.isEmpty()) {
+      const results = await (
+        await fetch(
+          `/adresses-register/match?municipality=${suggestion.municipality}&zipcode=${suggestion.zipCode}&thoroughfarename=${suggestion.street}&housenumber=${suggestion.housenumber}`
+        )
+      ).json();
+      addresses = results.map(function (result) {
+        return new Address({
+          uri: result.identificator.id,
+          addressRegisterId: result.identificator.objectId,
+          fullAddress: result.volledigAdres.geografischeNaam.spelling,
+          street: suggestion.street,
+          housenumber: suggestion.housenumber,
+          busNumber: result.busnummer ? result.busnummer : null,
+          zipCode: suggestion.zipCode,
+          municipality: suggestion.municipality,
+        });
+      });
+    }
+    return addresses;
+  }
+
+  /**
+   * Changes an address into an address suggestion
+   *
+   * @param {Address} address The address to transform into an addres suggestion
+   * @returns {AddressSuggestion}
+   */
+  toAddressSuggestion(address) {
+    return new AddressSuggestion({
+      street: address.street,
+      housenumber: address.number,
+      busNumber: address.boxNumber,
+      zipCode: address.postcode,
+      municipality: address.municipality,
+      fullAddress: address.fullAddress,
+    });
+  }
```

Comment:
Do we need this method? I'm not sure what the real difference between the 2 classes is? 1 has a uri and the other doesn't?

Reply:
We seem to use it in the frontends, but if we get rid of the classes, we might not need it anymore; And yes I think the uri is the only difference